utils/pool.go

```diff
@@ -0,0 +1,35 @@
+package utils
+
+import (
+	"math"
+	"sync"
+)
+
+type WaitGroupPool struct {
```

Comment:
the exported function should have comments (all the following functions need comments)

Reply:
done!